### PR TITLE
Fix build and add GFM tasklists

### PR DESCRIPTION
**Pull request by mpcjanssen**

### CODE REVIEW

File: generic/tclcmark.c [hunk 1]
```diff
@@ -244,6 +246,12 @@ static int tclcmark_render_cmd(
             goto vamoose;
     }
 
+    if (gfm || tasklist) {
+        fprintf(stderr, "Using tasklist extensions");
```
Comment (apnadkarni) [inline generic/tclcmark.c:250]: Why is this fprintf here? Left over?

**Review by apnadkarni — COMMENTED** [review]
Regarding the build issues, were these changes needed because cmark-gfm has changed? I'm pretty sure I'd built and tested the TEA based build.

Is that also why the -safe option was changed to -unsafe (because cmark-gfm changed it from the original cmark?)

Thanks for the patches.

**Comment by mpcjanssen** [comment]
Yep, gfm changed the default to safe (it's specifically mentioned as a difference with upstream)
The other changes were needed to make the code compile. These changes were probably also introduced by cmark-gfm

File: generic/tclcmark.c (hunk 1 above)
Comment (mpcjanssen) [reply generic/tclcmark.c:250]: yep sorry. I wasn't sure why "- [ ] abc" didn't work, which was silly driver error :).

**Pull request merged**